### PR TITLE
Bakta 1.6.1 fix error related to issue 5046

### PR DESCRIPTION
**Pull request by pimarin**
Due to some trouble in the bakta tool provided on galaxy,  I try to solve the problem 
Issue is [5046](https://github.com/galaxyproject/tools-iuc/issues/5046).

- I modified symlink related to database path to bakta and amrfinderplus

- [X] Modification of bakta and macro xml files
- [X] update on testtoolshed
- [X] test on local galaxy and work only for `--gram="?"`

I failed to use deepsig conda package in dependencies to integrate the `--gram` option

### CODE REVIEW

File: tools/bakta/bakta.xml [hunk 1]
```diff
@@ -11,10 +11,9 @@
...
 
     <command detect_errors="aggressive"><![CDATA[
-        mkdir ./database_path &&
-        ln -s '$(input_option.bakta_db_select.fields.path)/'* database_path &&
-        ln -s '$(input_option.amrfinder_db_select.fields.path)' database_path &&
-
+        mkdir -p ./database_path/amrfinderplus-db &&
+        ln -s '$(input_option.bakta_db_select.fields.path)'/* database_path &&
+        ln -s '$(input_option.amrfinder_db_select.fields.path)/' database_path/amrfinderplus-db/latest &&
         bakta
```
Comment (bgruening) [inline tools/bakta/bakta.xml:17]: This looks good to me, I have something similar locally.

File: tools/bakta/bakta.xml [hunk 2]
```diff
@@ -59,9 +58,6 @@
         #if $annotation.translation_table
             --translation-table '$annotation.translation_table'
         #end if
-        #if $annotation.gram
```
Comment (bgruening) [inline tools/bakta/bakta.xml:62]: Can you tell me which problem you had with deepsig? I guess we should fix this in the conda package and leave this options here?

Comment (oschwengers) [reply tools/bakta/bakta.xml:62]: DeepSig is really a tough one. I had my "fun" to finally get it into Bioconda. It has a dependency to a fairly old version of Tensorflow (=2.2.0) and a tight dependency to Python 3.8 (`python =3.8, <3.9`). There are many internal deps to older APIs why an easy and fast update won't be possible.

Therefore, I excluded the DeepSig dependency from the Bakta package, and hence I suggest to simply exclude the signal peptide prediction feature on Galaxy, as well.

**Comment by pimarin** [comment]
Test seem to be failed due to long time before starting but it's not possible to change it ?

File: tools/bakta/bakta.xml (hunk 2 above)
Comment (pimarin) [reply tools/bakta/bakta.xml]: I solved the problem by fixing some dependency, but time to conda build seem to be longer than expected for the github CI

**Comment by oschwengers** [comment]
@pimarin @bgruening @bernt-matthias , thanks for all your efforts and patience to work on this.

As discussed above, the manifold and older dependencies of DeepSig are becoming harder and harder to maintain on the software side and to solve on the dependency side.

May I therefore once suggest to postpone the `deepsig` related issues to a distinct PR? By doing so, we could feed the fixes for the `AMRfinder` and database related issues so that people can start using Bakta on Galaxy?

After this and in a dedicated own PR, we could work trying to get deepsig running. Thinking on the usability of the Bakta tool in Galaxy, I'd heavily argument to do so.

Hence, could we please exclude Deepsig, pinpoint `--gram` to `?`, focus on the db wrappers to polish the naming/versions and get it out?
Thanks again and best regards.

**Comment by pimarin** [comment]
I remove the `--gram` option, I can't solve the dependency resolution by conda to use deepsig, numpy and bakta

**Comment by pimarin** [comment]
So I fixed the --gram option on "?" and for next release if we have a stable solution I can change it ?

**Comment by oschwengers** [comment]
> So I fixed the --gram option on "?" and for next release if we have a stable solution I can change it ?

Yes, thanks a lot! This way, people are hopefully able to use Bakta in Galaxy very soon. All deepsig related issues are of rather minor importance as this only affects the signal prediction and only if this is requested by the user anyways.

File: tools/bakta/macro.xml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 
 <macros>
+    <token name="@TOOL_VERSION@">1.6.1</token>
+    <token name="@TOOL_VERSION@">1.6.1</token>
```
Comment (oschwengers) [inline tools/bakta/macro.xml:4]: These lines are doubled/tripled

**Comment by pimarin** [comment]
I update the bakta on the test toolshed, I tested all and It's fine to me

**Comment by pimarin** [comment]
I tested on test toolshed it's okay to me

File: tools/bakta/test-data/TEST_1/TEST_1.embl [hunk 4]
```diff
@@ -8,15 +8,15 @@ OS   .
 OC   .
 XX
 CC   Annotated with Bakta
-CC   Software: v1.5.0
```
Comment (bernt-matthias) [inline tools/bakta/test-data/TEST_1/TEST_1.embl:11]: Can you revert changes to modified tests data? Should be covered by `lines_diff`.

Comment (pimarin) [reply tools/bakta/test-data/TEST_1/TEST_1.embl:11]: sure

Comment (bernt-matthias) [reply tools/bakta/test-data/TEST_1/TEST_1.embl:11]: I meant to revert completely, i.e. that we also have `Software: v1.5.0` again :)

Comment (pimarin) [reply tools/bakta/test-data/TEST_1/TEST_1.embl:11]: i update test file with planemo and it's now 1.6?1 no ?

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks a lot. As usual awesome work @pimarin and @bernt-matthias!

File: tools/bakta/tool_data_table_conf.xml.test [hunk 5]
```diff
@@ -7,5 +7,6 @@
     <table name="amrfinderplus_database" comment_char="#">
         <columns>value, name, path</columns>
         <file path="${__HERE__}/test-data/amrfinderplus_database.loc" />
+
```
Comment (bernt-matthias) [inline tools/bakta/tool_data_table_conf.xml.test:10]:
```suggestion
```

**Comment by bernt-matthias** [comment]
Let me take care of the test data :) 


Can you tell us if there is anything else that needs to be done to the datamanager?

**Comment by pimarin** [comment]
I tested the data manager on testoolshed and i have the version number for user instead of the commit number so to me it's great

**Comment by pimarin** [comment]
thank to your help @bernt-matthias

**Comment by oschwengers** [comment]
Great work! Thanks @pimarin @bernt-matthias and @bgruening !!!

**Comment by pimarin** [comment]
Thank a lot, I hope it works fine now, I will install it rapidly to my lab on their galaxy instance !

**Comment by pimarin** [comment]
I changed it to have the version instead of the zenodo doi. It worked when I tested locally

**Pull request merged**